feg/gateway/sbi/sbi_test.go

```
@@ -0,0 +1,165 @@
+
+	// Create a new SM Policy
+	ipAddr := sbi.Ipv4Addr(IP_ADDR1)
+	imsi := sbi.Supi(IMSI1)
+	body := n7_client.PostSmPoliciesJSONRequestBody{
+		Ipv4Address:  &ipAddr,
+		PduSessionId: 10,
+		Supi:         imsi,
+	}
+
+	resp, err := client.PostSmPolicies(context.Background(), body)
+	assert.NoError(t, err, "Failed to create SM Policies")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	// Get SM Policy given the policyId
+	policyResp, err := client.GetSmPoliciesSmPolicyIdWithResponse(context.Background(), POLICY_ID1)
+	assert.NoError(t, err, "Failed to get SM Policies")
+	assert.Equal(t, http.StatusOK, policyResp.StatusCode())
+	assert.Equal(t, sbi.Ipv4Addr(IP_ADDR1), *(policyResp.JSON200.Context.Ipv4Address))
+	assert.Equal(t, sbi.Supi(IMSI1), policyResp.JSON200.Context.Supi)
+}
+
+func startServer(t *testing.T) *MockPcf {
+	errChan := make(chan error)
+	eserver := echo.New()
+
+	mockPcf := NewMockPcf()
+	go func() {
+		n7.RegisterHandlers(eserver, mockPcf)
+
+		err := eserver.Start("localhost:0")
+		if err != nil {
+			errChan <- err
+		}
+	}()
+
+	addr, err := waitForTestServer(eserver, errChan)
+	assert.NoError(t, err)
+	mockPcf.ListenAddr = addr
+
+	return mockPcf
+}
+
+// waitForTestServer waits for the Echo server to be launched and returns the Listener Address.
+// There is a similar one in magma/orc8r/cloud/go/obsidian/access/tests but that uses a older
+// version of Echo server.
+func waitForTestServer(e *echo.Echo, errChan <-chan error) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 200*time.Millisecond)
+	defer cancel()
+
+	ticker := time.NewTicker(5 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return "", ctx.Err()
+		case <-ticker.C:
+			addr := e.ListenerAddr()
+			if addr != nil && strings.Contains(addr.String(), ":") {
+				// server started
+				return addr.String(), nil
+			}
+		case err := <-errChan:
+			if err == http.ErrServerClosed {
+				return "", nil
```

Why is this function taking an error and not returning it? This seems to be hiding an error

---

This is the error that will be returned when there is a graceful shutdown of the server. For example when the test case shuts down the server due to some other error.

---

Ok looks fine, lets leave it as it is.
I am still not sure we need that case. It looks like that case may be useful for echo testing but not for us. But lets leave it as it is